src/Compilers/Core/Portable/SourceGeneration/SourceGeneratorContext.cs

```diff
@@ -59,7 +60,14 @@ internal SourceGeneratorContext(Compilation compilation, ImmutableArray<Addition
 
         internal AdditionalSourcesCollection AdditionalSources { get; }
 
-        public void AddSource(string hintName, SourceText sourceText) => AdditionalSources.Add(hintName, sourceText);
+        public void AddSource(string hintName, SourceText sourceText)
+        {
+            if (!Path.GetExtension(hintName).Equals(".cs", StringComparison.OrdinalIgnoreCase))
```

Comment:
`GetExtension` can return `null`

Reply:
Hmm, only when hintName is `null` 

---
In reply to: [434195700](https://github.com/dotnet/roslyn/pull/44768#discussion_r434195700) [](ancestors = 434195700)